bundles/org.jkiss.utils/src/org/jkiss/utils/ArrayUtils.java

```diff
@@ -259,4 +256,14 @@ public static void main(String[] args) {
     public static <T> T[] toArray(Class<T> type, Collection<? extends T> list) {
         return list.toArray((T[]) Array.newInstance(type, list.size()));
     }
+
+    public static String[] getFilteredStringArray(String[] array) {
+        List<String> filteredList = new ArrayList<>();
+        for (String token : array) {
+            if (CommonUtils.isNotEmpty(token)) {
+                filteredList.add(token);
+            }
+        }
+        return filteredList.toArray(new String[0]);
+    }
```

Comment:
Let's refactor this function to be useful in other cases:

```java
    @NotNull
    public static <T> T[] filter(T[] array, Predicate<T> predicate) {
        List<String> filteredList = new ArrayList<>();
        for (String item : array) {
            if (predicate.test(token)) {
                filteredList.add(token);
            }
        }
        return filteredList.toArray(new String[0]);
    }
```

Also, I would suggest pre-allocating `filteredList` to at most the size of input `array` and then making a required slice of it.

The problem is that by generalizing it, it's just a dummy way to write:

```java
Arrays.stream(...).filter(...).toArray(T[]::new)
```

Reply:
Maybe streams are in a general better solution.